tap_github/client.py

```diff
@@ -0,0 +1,271 @@
+import time
+import requests
+import backoff
+from simplejson import JSONDecodeError
+import singer
+from singer import metrics
+
+LOGGER = singer.get_logger()
+DEFAULT_SLEEP_SECONDS = 600
+MAX_SLEEP_SECONDS = DEFAULT_SLEEP_SECONDS
```

Comment:
@prijendev In the previous logic, MAX_SLEEP_SECONDS was taken from config parameter "max_sleep_seconds" if provided. Why is that logic not present here?
Reference - https://github.com/singer-io/tap-github/blob/4f7ba58e501bd5026979194a0c25f31d32d3ffe0/tap_github/__init__.py#L1184